src/nodes/utils/Packing.js

```diff
@@ -19,3 +20,25 @@ export const directionToColor = ( node ) => nodeObject( node ).mul( 0.5 ).add( 0
  * @return {Node<vec3>} The direction.
  */
 export const colorToDirection = ( node ) => nodeObject( node ).mul( 2.0 ).sub( 1 );
+
+/**
+ * Unpacks RG-packed normals by reconstructing their Z component by projecting it to the hemisphere.
+ * The XY coordinates of the input normal are expected to be in the [-1, 1] range.
+ *
+ * @tsl
+ * @function
+ * @param {Node<vec3>} normal - The packed normal to unpack.
+ * @return {Node<vec3>} The unpacked normal.
+ */
+export const unpackRGNormal = ( normal ) => vec3( normal.xy, sqrt( saturate( float( 1.0 ).sub( dot( normal.xy, normal.xy ) ) ) ) );
```

Comment:
Imo this should be called reconstructZ and it should be channel ordering unaware. I.e. you should pass normal x,y yourself.

Reply:
How about this:

```
export const reconstructZ = ( xy ) => vec3( xy, sqrt( saturate( float( 1.0 ).sub( dot( xy, xy ) ) ) ) );
```

Since this is generating a syntax tree, I'd rather pass a vec2 than two separate x,y values, as that will result in cleaner shader code generation.